### DISCUSSION TITLE
Malformed json response in API endpoint

### DISCUSSION THREAD

**Discussion by dorbal** [vercel/next.js#27891] [Help]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

14.17.1

### What browser are you using?

Chrome

### What operating system are you using?

Linux

### How are you deploying your application?

next dev, next start

### Describe the Bug

I've got "SyntaxError: Unexpected token / in JSON at position 0":

res.status("200").json("string");
result: string 

### Expected Behavior

res.status("200").json("string");

The result should be: "string"

### To Reproduce

Just make an API route with the following content:

export default function handler(req, res) {
  res.status("200").json("string");
}

**Comment by rishimohan** [comment]
@dorbal Try `res.status(200).json({result: "string"});`

Notice the status code `200`, it shouldn't be in quotes.

**Comment by dorbal** [comment]
Anyway. Is it a bug or not? Just because it's closed without any feedback.

**Reply by rishimohan** [reply]
It's definitely not a bug

**Reply by dorbal** [reply]
Can you explain it for me, why?

**Reply by rishimohan** [reply]
The status is expected to be an integer and not string, you're passing `"200"` which is a string, you need to pass `200`.

**Reply by dorbal** [reply]
Sorry we misunderstood each other. 

I'm talking about the .json("string") part of the code, and not the res.status(200) part. 

This code:
res.status(200).json("foobar");
Produces text response, not json encoded string.

I mean the result is foobar without quotation marks.

**Reply by rishimohan** [reply]
My bad, the json should be an object, you can't do `.json("foobar")` which means you're passing string and not object, it should be something like `.json({"foobar"})` or `.json({result: "foobar"})`

**Reply by dorbal** [reply]
I see.

According to the documentation (https://nextjs.org/docs/api-routes/response-helpers):
res.json(body) - Sends a JSON response. body must be a serialiazable object.

In my mind the string type is an object as well. 

I've tried this:
res.status(200).json(new String("foobar"));
And it works.

But, why is this restriction exist? Why I cannot use any type that have json representation?

I assumed that if I call the json() method I'll get the same as JSON.stringify().

**Reply by rishimohan** [reply]
You can change the type of data this API sends using HTTP headers, ref: https://www.geeksforgeeks.org/http-headers-content-type/

**Reply by dorbal** [reply]
The problem is not about any http header (status code, content type etc.), the problem is the wrong http body.

Take a look at this handler:
```javascript
const data = "this is a string"

export default function handler1(req, res) {
  res
    .status(200)
    .setHeader("Content-Type", "application/json; charset=utf-8")
    .send(JSON.stringify(data));
}
```
It sends "application/json" content type header, and the body is ***always*** well formed json. 


On the other hand, this handler is buggy:
```javascript
const data = "this is a string"

export default function handler1(req, res) {
  res.status(200).json(data)
}
```
It sends "applicaion/json" content type header as well, but sometimes the body is not json. It works if the data variable contains object. However, if the data variable contains string, the body will be *just* string.

I think, this is a bug without warning. It will turn out only if you try to parse the http body as json on client side. (You will get "SyntaxError: Unexpected token / in JSON at position 0" error).
